packages/shared/src/components/onboarding/headline/PersonalizedOnboardingHeadline.tsx

```diff
@@ -0,0 +1,103 @@
+import type { ReactElement } from 'react';
+import React from 'react';
+import dynamic from 'next/dynamic';
+import classNames from 'classnames';
+import { useQuery } from '@tanstack/react-query';
+import { useSearchParams } from 'next/navigation';
+import { useAuthContext } from '../../../contexts/AuthContext';
+import {
+  getSocialReferrer,
+  socialCTA,
+  socialGradient,
+  socialIcon,
+  SocialIconType,
+} from '../../../lib/socialMedia';
+import { geoToEmoji, geoToCountry } from '../../../lib/geo';
+import { capitalize } from '../../../lib/strings';
+import { IconSize } from '../../Icon';
+import { getBasicUserInfo } from '../../../graphql/users';
+import { generateQueryKey, RequestKey } from '../../../lib/query';
+import { ProfilePicture } from '../../ProfilePicture';
+
+const OnboardingHeadline = dynamic(() =>
+  import('../../auth/OnboardingHeadline').then((mod) => mod.OnboardingHeadline),
+);
+
+const PersonalizedOnboardingHeadline = (): ReactElement => {
+  const { geo } = useAuthContext();
+  const searchParams = useSearchParams();
+  const userId = searchParams?.get('userid');
+  const { data: user, isError } = useQuery({
+    queryKey: generateQueryKey(RequestKey.ReferringUser, { id: userId }),
+    queryFn: () => getBasicUserInfo(userId),
+    enabled: !!userId,
+    retry: false,
+  });
+
+  if (userId && !isError) {
+    const name = user?.name ? user?.name.split(' ')[0] : user?.username;
```

Comment:
```suggestion
    const name = user?.name?.split(' ')[0] || user?.username;
```

Reply:
Might also be nice if we can take this out and make it a reusable function to fetch a single word name.